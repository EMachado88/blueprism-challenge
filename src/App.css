/* Basics */

button {
  background-color: #898fa3;
  border: 3.5px solid #898fa3;
  border-radius: 3.5px;
  text-transform: uppercase;
  letter-spacing: 1px;
  font-weight: bolder;
  font-size: 16px;
  padding: 7px;
  cursor: pointer;

  -webkit-transition: all 0.2s 0s ease-out;
  -moz-transition: all 0.2s 0s ease-out;
  -o-transition: all 0.2s 0s ease-out;
  transition: all 0.2s 0s ease-out;
}

button:hover,
button:active {
  background-color: #282c34;
  border-radius: 3.5px;
  text-transform: uppercase;
  font-weight: bolder;
  font-size: 16px;
  color: #898fa3;
  padding: 7px;
  cursor: pointer;
}

select,
input {
  width: 100%;
  padding: 10px;
  font-size: 16px;
}

/* Base app */

.App {
  height: 100vh;
  display: flex;
  flex-direction: column;
  background-color: #898fa3;
  color: #282c34;
}

.App > * {
  padding: 30px;
}

.App-header {
  background-color: #282c34;
  font-size: 32px;
  font-weight: bolder;
  color: #fff;
  display: flex;
  justify-content: space-between;
  align-items: center;
  box-shadow: 5px 5px 15px 1px rgba(0,0,0,0.25);
}

.menu {
  cursor: pointer;
}

.main {
  display: flex;
  height: 100%;
  overflow: hidden;
}

@media only screen and (max-width: 1023px) {
  .main {
    flex-direction: column;
  }
}

.main > * {
  overflow: auto;
}

/* Schedules */

.schedules {
  margin-right: 30px;
  padding-right: 30px;
  min-width: 400px;
}

@media only screen and (max-width: 1023px) {
  .schedules {
    height: 370px;
    margin-bottom: 30px;
    display: flex;
  }

  .schedules > * {
    min-width: 300px;
    margin-right: 30px !important;
  }
}

.schedule {
  background-color: #282c34;
  color: #fff;
  border-radius: 7px;
  margin: 10px 0;
  padding: 20px;
  cursor: pointer;
  box-shadow: 5px 5px 15px 1px rgba(0,0,0,0.25);

  -webkit-transition: all 0.2s 0s ease-out;
  -moz-transition: all 0.2s 0s ease-out;
  -o-transition: all 0.2s 0s ease-out;
  transition: all 0.2s 0s ease-out;
}

.schedule:hover {
  transform: translate3d(10px, 0, 0);
}

.schedule:active,
.schedule.active {
  transform: translate3d(10px, 0, 0);
  background-color: #4a6299;
}

.schedule:first-child {
  margin-top: 0;
}

.schedule:last-child {
  margin-bottom: 0;
}

.schedule .schedule-description {
  margin-bottom: 15px;
}

.schedule .schedule-button {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

/* Logs */

.logs {
  flex-grow: 1;
  display: flex;
  flex-wrap: wrap;
}

.empty-logs {
  display: flex;
  flex-grow: 1;
  align-items: center;
  justify-content: center;
  height: 100%;
}

.log {
  border: 3.5px solid #282c34;
  border-radius: 7px;
  margin-right: 30px;
  margin-bottom: 30px;
  padding: 20px;
  width: calc(25% - 30px);
  flex-wrap: wrap;
  box-shadow: 5px 5px 15px 1px rgba(0,0,0,0.25);

  -webkit-transition: all 0.2s 0s ease-out;
  -moz-transition: all 0.2s 0s ease-out;
  -o-transition: all 0.2s 0s ease-out;
  transition: all 0.2s 0s ease-out;
}

@media only screen and (max-width: 1399px) {
  .log {
    width: calc(33.33% - 30px);
  }
}

@media only screen and (max-width: 1279px) {
  .log {
    width: calc(50% - 30px);
  }
}

.log > p {
  display: flex;
  justify-content: space-between;
}

.log:hover {
  transform: translate3d(10px, 0, 0);
}
