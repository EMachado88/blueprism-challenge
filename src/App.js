import './App.css'

import Header from './components/Header'
import Log from './components/Log'
import Schedule from './components/Schedule'

import axios from 'axios'
import { useCallback, useEffect, useState } from 'react'

const API_URL = 'http://localhost:3000'

const App = () => {
  const [schedules, setSchedules] = useState([])
  const [selectedSchedule, setSelectedSchedule] = useState(0)
  const [logs, setLogs] = useState([])
  const [filteredLogs, setFilteredLogs] = useState([])

  useEffect(() => {
    const fetchSchedules = async () => {
      const { data } = await axios.get(`${API_URL}/schedules`)
      setSchedules(data)
    }

    const fetchLogs = async () => {
      const { data } = await axios.get(`${API_URL}/scheduleLogs`)
      setLogs(data)
    }

    fetchSchedules()
    fetchLogs()
  }, [])

  const handleClickSchedule = useCallback(
    (id) => () => {
      const filteredLogs = logs.filter(log => log.scheduleId === id)

      setFilteredLogs(filteredLogs)
      setSelectedSchedule(id)
    },
    [],
  )
  
  return (
    <div className="App">
      <Header />

      <main className='main'>
        <div className='schedules'>
          {schedules.map(schedule => (
            <Schedule
              key={schedule.id}
              schedule={schedule}
              handleClickSchedule={handleClickSchedule}
            />
          ))}
        </div>
        <div className='logs'>
          {filteredLogs.map((log, index) => (
            <Log
              key={log.id}
              log={log}
            />
          ))}
        </div>
      </main>
    </div>
  );
}

export default App
