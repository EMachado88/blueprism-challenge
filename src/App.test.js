import renderer from 'react-test-renderer'

import App from './App'

test('Renders App correctly', () => {
  const tree = renderer
    .create(
      <App />,
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
});
