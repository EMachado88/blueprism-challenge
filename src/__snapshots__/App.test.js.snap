// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders App correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <span>
      Schedules
    </span>
    <span
      className="material-icons menu"
    >
      menu
    </span>
  </header>
  <main
    className="main"
  >
    <div
      className="schedules"
    >
      <div
        className="filters mb-30"
      >
        <h3
          className="uppercase mb-10 d-flex justify-between"
        >
          <span>
            Filters
          </span>
          <span
            className="material-icons menu"
          >
            tune
          </span>
        </h3>
        <select
          className="mb-10"
          onChange={[Function]}
        >
          <option
            value=""
          >
            Interval [any]
          </option>
        </select>
        <input
          className="mb-10"
          id="searchQuery"
          onChange={[Function]}
          placeholder="Search by name or description"
          type="text"
        />
        <div
          className="d-flex mb-20"
        >
          <div
            className="d-flex w-50"
          >
            <input
              checked={true}
              id="retired"
              name="retired"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor="retired"
            >
              <strong>
                Retired
              </strong>
            </label>
          </div>
          <div
            className="d-flex w-50"
          >
            <input
              checked={true}
              id="unretired"
              name="unretired"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor="unretired"
            >
              <strong>
                Unretired
              </strong>
            </label>
          </div>
        </div>
        <hr />
      </div>
    </div>
    <div
      className="logs"
    >
      <div
        className="empty-logs"
      >
        <h1>
          Select a schedule to view its logs
        </h1>
      </div>
    </div>
  </main>
</div>
`;
